### PR TITLE
Add ability for caller to override quantity display format per QuantityType.

### PR DESCRIPTION
**Pull request by bsteinbk**
This PR contains a sub-set of PR #326 since these changes are needed as soon as possible. This PR does not contain the CustomFormatter implementation added by the Civil group. The Civil group have some changes they want to make to that code before merging with master.

### CODE REVIEW

**Review by pmconne — CHANGES_REQUESTED** [review]
Why `any`?

**Review by DanEastBentley — APPROVED** [review]

**Review by pmconne — APPROVED** [review]

**Review by calebmshafer — CHANGES_REQUESTED** [review]

File: common/api/imodeljs-frontend.api.md [hunk 1]
```diff
@@ -6414,11 +6423,15 @@ export class QuadId {
     row: number;
 }
 
-// @alpha
+// @beta
```
[the remark this answers is not in the document]
Comment (bsteinbk) [reply common/api/imodeljs-frontend.api.md]: Reverted tag back to alpha. Also add SurveyFeet test to mimic native unit conversion test.

File: core/quantity/src/Formatter/Format.ts [hunk 2]
```diff
@@ -419,39 +418,38 @@ export class Format implements FormatProps {
...
-      composite.spacer = this.spacer;
-      composite.includeZero = this.includeZero;
-      composite.units = [];
-      this.units.forEach((unit: any) => {
-        if (undefined !== unit[1])
-          composite.units.push({
-            name: unit[0].name,
-            label: unit[1],
-          });
+  public toJson(): FormatProps {
```
Comment (calebmshafer) [inline core/quantity/src/Formatter/Format.ts:421]: At some point we probably should rename name this to `toJSON` for consistency with how we use it in other places.  However, not needed in this PR.

**Review by calebmshafer — APPROVED** [review]

**Comment by calebmshafer** [comment]
@mergifyio backport release/2.10.0

**Pull request merged**